client/components/layout/MainLayout.tsx

```diff
@@ -30,90 +13,17 @@ import '@client/styles/MainLayout.css';
  * - The `Outlet` component handles rendering of route-specific content in the main section.
  */
 const MainLayout: React.FC = () => {
-  // const { showTopNavBar, showFooter } = useDynamicNavigation();
-  const [open, setOpen] = useState(true);
-  const toggleDrawer = (newOpen: boolean) => () => {
-    setOpen(newOpen);
-  };
-  const mountedStyle = { animation: 'opacity-in 300ms ease-in' };
-  const unmountedStyle = {
-    animation: 'opacity-out 200ms ease-in',
-    animationFillMode: 'forwards',
-  };
-
-  const DrawerList = (
-    <Box className="side-nav-bar">
-      <Box className="linkta-logo-drawer">
-        <img
-          className="linkta-drawer-image"
-          src="../assets/linkta-logo-transparent.svg"
-          alt="The Linkta.io logo at approximately 50 pixels square, located in the upper-left corner of the page."
-        />
-      </Box>
-
-      <Link
-        className="side-nav-link"
-        underline="none"
-        mt={3}
-        component={RouterLink}
-        to="/sign-in"
-      >
-        <AccountCircleOutlined />
-        <Typography variant="caption">test.user@linkta.org</Typography>
-      </Link>
-
-      <Link
-        className="side-nav-link"
-        underline="none"
-        component={RouterLink}
-        to="/generate"
-      >
-        <AddCircleOutline />
-        <Typography variant="caption">Explore a New Topic</Typography>
-      </Link>
-
-      <Box
-        className="recent-user-inputs"
-        mt={5}
-        pl={2}
-      >
-        <Typography
-          variant="body2"
-          color={'primary.contrastText'}
-        >
-          Recent
-        </Typography>
-      </Box>
-
-      <Link
-        className="side-nav-link"
-        underline="none"
-        mt={'auto'}
-        component={RouterLink}
-        to="/help-and-feedback"
-      >
-        <HelpOutlineOutlined />
-        <Typography variant="caption">Help and Feedback</Typography>
-      </Link>
+  const { width } = useMyViewport();
+  const breakpoint = 768;
 
-      <Link
-        className="side-nav-link"
-        underline="none"
-        component={RouterLink}
-        to="/settings"
-      >
-        <SettingsOutlined />
-        <Typography variant="caption">Settings</Typography>
-      </Link>
+  const [drawerOpen, setDrawerOpen] = useState(width > breakpoint);
+  const toggleDrawer = () => {
+    width > breakpoint && setDrawerOpen(!drawerOpen);
+  };
 
-      <Button
-        className="drawer-close-button"
-        onClick={toggleDrawer(false)}
-        startIcon={<ChevronLeftOutlined />}
-        sx={{ marginBottom: '20px', paddingInline: '20px' }}
-      ></Button>
-    </Box>
-  );
+  useEffect(() => {
+    width < breakpoint && setDrawerOpen(false);
+  }, [width < breakpoint && width]);
```

Comment:
I've never seen an expression in a dependency array before, and I'm pretty sure we only want to put variables stored in React state in the depndency array.  Also, I noticed that the SideNavDrawer was not popping open when expanding the size of the window.  This seemed to fix the problem: 

```
  useEffect(() => {
    width < breakpoint ? setDrawerOpen(false) : setDrawerOpen(true);
  }, [width]);
  ```

Reply:
@colinvandergraaf This may become irrelevant with the switch from window.resize to matchMedia.

I had a brief debate on this topic with Chris and Mark on the CS alumni channel. The tl;dr is I found a few articles and discussions that support the validity of this pattern:
https://codesmithalumni.slack.com/archives/C03H6P87MNJ/p1717314849714749

The inspiration for this was creating the behavior where the side nav closes and remains compact while the view width is less than the breakpoint value, and the side nav does not automatically open when the view width is greater than the breakpoint value. I may have a large screen, and I may still prefer the compact nav bar. 

https://www.reddit.com/r/reactjs/comments/197ltkf/is_it_ok_to_use_a_conditional_as_a_useeffect/
https://dev.to/rangeoshun/optimize-useeffect-by-using-the-condition-itself-as-a-dependency-1c2j
https://stackoverflow.com/questions/61439712/react-conditional-dependency-on-useeffect-or-how-to-prevent-redundancy

Reply:
Thanks for sharing the thread and the articles. good to know.  

Re: the SideNav. It's true that when expanding the size of the window some users may prefer a collapsed SideNav maybe the better question is what should our default behavior be? IMO, it seems more common to have the SideNav to auto open on expansion.  That's the behavior on YouTube and MDN. But it does seem counter intuitive to a degree when you're typically expanding the window to have more room to work with the primary content of the page

Reply:
I think that behavior is far more common, and that's because it's common to build responsive mobile-friendly layouts. I think this is different because the app is not intended to be mobile-friendly. But yes, these questions should be asked and the answers should be in the Figma doc.